linkerd/distribute/src/service/empty.rs

```diff
@@ -0,0 +1,38 @@
+use linkerd_stack::Service;
+use std::{
+    marker::PhantomData,
+    task::{Context, Poll},
+};
+
+#[derive(Debug, Copy, Clone)]
+pub struct EmptySelection<S> {
+    _boo: PhantomData<fn() -> S>,
+}
+
+impl<S> Default for EmptySelection<S> {
+    fn default() -> Self {
+        Self {
+            _boo: Default::default(),
+        }
+    }
+}
+
+impl<Req, S> Service<Req> for EmptySelection<S>
+where
+    S: Service<Req>,
+{
+    type Response = S::Response;
+    type Error = S::Error;
+    type Future = S::Future;
+
+    #[inline]
+    fn poll_ready(&mut self, _: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+        tracing::debug!("empty distribution will never become ready");
+        Poll::Pending
+    }
+
+    #[inline]
+    fn call(&mut self, _: Req) -> Self::Future {
+        unreachable!("EmptySelection is never ready")
+    }
+}
```

Comment:
This type doesn't really carry any weight -- we can easily collapse this case into the parent service

Reply:
Done